File: 1800-maximum-ascending-subarray-sum/1800-maximum-ascending-subarray-sum.py
class Solution:
    def maxAscendingSum(self, nums: List[int]) -> int:
        n = len(nums)
        maxSum = nums[0]

        tempSum = nums[0]
        for i in range(1, n):
            if(nums[i] > nums[i-1]):
                tempSum += nums[i]
                maxSum = max(maxSum, tempSum)
            else:
                tempSum = nums[i]
        return maxSum

File: 1800-maximum-ascending-subarray-sum/1800-maximum-ascending-subarray-sum.cpp
class Solution {
public:
    int maxAscendingSum(vector<int>& nums) {
        int n = nums.size();
        int maxSum = nums[0];

        int tempSum = nums[0];
        for(int i=1; i<n; i++) {
            if(nums[i] > nums[i-1]) {
                tempSum += nums[i];
                maxSum = max(maxSum, tempSum);
            }
            else {
                tempSum = nums[i];
            }
        }
        return maxSum;
    }
};